test_conformance/api/test_queries.cpp

```diff
@@ -244,25 +253,107 @@ int test_get_sampler_info(cl_device_id deviceID, cl_context context, cl_command_
+            }
+        }
+    }
     return 0;
 }
+int test_get_sampler_info(cl_device_id deviceID, cl_context context,
+                          cl_command_queue queue, int num_elements)
+{
+    int error;
+    PASSIVE_REQUIRE_IMAGE_SUPPORT(deviceID)
 
-#define TEST_COMMAND_QUEUE_PARAM( queue, paramName, val, expected, name, type, cast )    \
-error = clGetCommandQueueInfo( queue, paramName, sizeof( val ), &val, &size );        \
-test_error( error, "Unable to get command queue " name );                            \
-if( val != expected )                                                                \
-{                                                                                    \
-log_error( "ERROR: Command queue " name " did not validate! (expected " type ", got " type ")\n", (cast)expected, (cast)val );    \
-return -1;                                                                        \
-}            \
-if( size != sizeof( val ) )                \
-{                                        \
-log_error( "ERROR: Returned size of command queue " name " does not validate! (expected %d, got %d)\n", (int)sizeof( val ), (int)size );    \
-return -1;    \
+    error = get_sampler_info_params(deviceID, context, false);
+    test_error(error, "Test Failed");
+
+    return 0;
+}
+
+int test_get_sampler_info_compatibility(cl_device_id deviceID,
+                                        cl_context context,
+                                        cl_command_queue queue,
+                                        int num_elements)
+{
+    int error;
+    PASSIVE_REQUIRE_IMAGE_SUPPORT(deviceID)
+
+    error = get_sampler_info_params(deviceID, context, true);
+    test_error(error, "Test Failed");
+
+    return 0;
+}
+
+template <typename T>
+int command_queue_param_test(cl_command_queue queue,
+                             cl_command_queue_info param_name, T expected,
+                             const char *name)
+{
+    size_t size;
+    T val;
+    int error =
+        clGetCommandQueueInfo(queue, param_name, sizeof(val), &val, &size);
+    test_error(error, "Unable to get command queue info");
+    if (val != expected)
+    {
+        test_fail("ERROR: Command queue %s did not validate!\n", name);
+    }
+    if (size != sizeof(val))
+    {
+        test_fail("ERROR: Returned size of command queue %s does not validate! "
+                  "(expected %d, got %d)\n",
+                  name, (int)sizeof(val), (int)size);
+    }
+    return 0;
 }
 
-int test_get_command_queue_info(cl_device_id deviceID, cl_context context, cl_command_queue ignoreQueue, int num_elements)
+#define MIN_NUM_COMMAND_QUEUE_PROPERTIES 2
+#define OOO_NUM_COMMAND_QUEUE_PROPERTIES 4
+static cl_command_queue_properties property_options[] = {
+    0,
+    CL_QUEUE_PROFILING_ENABLE,
+    CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE,
+    CL_QUEUE_PROFILING_ENABLE,
+    CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE,
```

Comment:
I'm not sure I quite understand this array. Should this be `CL_QUEUE_PROFILING_ENABLE  | CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE`, rather than repeating those two properties individually?